app/src/main/java/ca/klapstein/baudit/models/DataModel.java

```diff
@@ -35,10 +32,8 @@ public DataModel(Context context) {
      * @param username {@code String}
      * @return {@code true} if the username {@code String} representation is not already taken, otherwise {@code false}
      */
-    static public boolean uniqueID(String username) {
-        // TODO: validate from the local?
-        // TODO: implement from RemoteModel
-        return true;
+    public boolean uniqueID(Username username) {
+        return getPatient(username) == null && getCareProvider(username) == null;
```

Comment:
We are allowing a patient and a care provider to have the same username?

Reply:
this checks that there exists no `Patient` with the given username

it also checks that there exists no `CareProvider` with the given username

either either a `CareProvider` or `Patient` exists with a username it is not a unique username

thus, two patient(s) and/or care provider(s) cannot share the same username.